### ISSUE TITLE
Mentored by two mentors, same exercise, UI is broken

### ISSUE THREAD

**Issue by joshgoebel** [exercism/exercism#6155]
I'm being mentored by Anas and Victor on a single exercise:

<img width="695" alt="Screen Shot 2021-12-13 at 12 03 07 AM" src="https://user-images.githubusercontent.com/6473/145755306-7843704d-ddc4-4144-bf98-fa03e7d97464.png">

The URLs are distinct:

- https://exercism.org/tracks/ruby/exercises/connect/mentor_discussions/76433efe-08c7-11ec-81c5-853579c25f94
- https://exercism.org/tracks/ruby/exercises/connect/mentor_discussions/76c8e946-08c7-11ec-81c5-853579c25f94

And the UI/header trim is unique also, but the message dialog is ALWAYS my conversation with `kotp`/Victor, even when I'm trying to work on my conversation with Anas.

<img width="625" alt="Screen Shot 2021-12-13 at 12 04 34 AM" src="https://user-images.githubusercontent.com/6473/145755389-2417f3b5-3194-4737-800a-36b2c2533f27.png">

**Comment by kotp** [comment]
It is an old exercise, though, so it inherits some things.  I have a lot of exercises like this in my queue (mentors queue) currently, I believe.

**Comment by iHiD** [comment]
@joshgoebel Yeah - it shouldn't be allowed. The code expects only one in-progress mentoring session at any time, and so behaves unpredictably in this situation. So the bug is that it was allowed in the first place.

**Comment by kotp** [comment]
Of interest, I do not have you in my queue awaiting or pending my response for any exercises @joshgoebel if I remember your alias there.  (Also those links do not work for me, though I expected they would not.)

**Comment by joshgoebel** [comment]
Even after ending both discussions... (the system also seemed confused if they were ended or not, yet it let me walk thru the end procedure and now both slots are free again)...

The bug remains and visiting them historically:

<img width="917" alt="Screen Shot 2021-12-13 at 4 18 51 PM" src="https://user-images.githubusercontent.com/6473/145890434-573f3db8-103d-421d-93b9-43701b1aa748.png">

This still results in the broken views where I can only ever find my dialog with Victor... both links leads to the VIctor discussion even if the header/UI says it's the other mentoring session (see above snaps).

This seems a bug that should be fixed even if it's only a data issue (some type of cleanup migration, etc).

**Comment by alkhatib-shopify** [comment]
I don't remember ever reviewing this exercise.
I also found a few more exercises that appear in my `finished` queue that I was not involved with at all.

https://exercism.org/mentoring/discussions/76ba337e-08c7-11ec-81c5-853579c25f94
https://exercism.org/mentoring/discussions/763e8d28-08c7-11ec-81c5-853579c25f94
https://exercism.org/mentoring/discussions/765f43c4-08c7-11ec-81c5-853579c25f94

**Comment by iHiD** [comment]
@alkhatib-shopify You say you weren't involved at all, but there are posts in those discussions that we think were written by you? For example on June 1st:

> Good work passing the tests. I'm approving your solution so you don't have to wait before progressing to new exercises. I would encourage you to check out Ruby's transpose method and see if you can use this to shorten the logic in your initialize method. Also I advise removing the comments - they are not part of your code and so should be deleted.

Did you not write that?

**Comment by alkhatib-shopify** [comment]
Which discussion in particular is that comment from?

e.g for https://exercism.org/mentoring/discussions/765f43c4-08c7-11ec-81c5-853579c25f94
This is the discussion that I see: 
![discussion](https://screenshot.click/21-12-j9s3m-ebu5t.png)

I don't see any comments from me.

**Comment by iHiD** [comment]
@alkhatib-shopify The quote is from: https://exercism.org/mentoring/discussions/76ba337e-08c7-11ec-81c5-853579c25f94 - You don't see any comments from you on that?

763e8d28-08c7-11ec-81c5-853579c25f94 is weird though. I think that you probably started mentoring that solution but never posted a comment, which in v2 meant that you still got a row saved in the db. These were supposed to be removed during ETL but it seems something's not worked correctly. I'd have to check in the old database to see for sure what's happened.

**Comment by alkhatib-shopify** [comment]
This is what I see for https://exercism.org/mentoring/discussions/76ba337e-08c7-11ec-81c5-853579c25f94:
![image](https://user-images.githubusercontent.com/68076293/146573887-43c9f849-bedc-4529-b42e-6075c4d8b579.png)
